static/app/components/events/eventReplay/replayClipPreview.tsx

```diff
@@ -0,0 +1,296 @@
+import {ComponentProps, Fragment, useMemo, useRef} from 'react';
+import styled from '@emotion/styled';
+import screenfull from 'screenfull';
+
+import {Alert} from 'sentry/components/alert';
+import {Button, LinkButton} from 'sentry/components/button';
+import ButtonBar from 'sentry/components/buttonBar';
+import ErrorBoundary from 'sentry/components/errorBoundary';
+import Placeholder from 'sentry/components/placeholder';
+import {Flex} from 'sentry/components/profiling/flex';
+import MissingReplayAlert from 'sentry/components/replays/alerts/missingReplayAlert';
+import ReplayTimeline from 'sentry/components/replays/breadcrumbs/replayTimeline';
+import {PlayerScrubber} from 'sentry/components/replays/player/scrubber';
+import useScrubberMouseTracking from 'sentry/components/replays/player/useScrubberMouseTracking';
+import {
+  Provider as ReplayContextProvider,
+  useReplayContext,
+} from 'sentry/components/replays/replayContext';
+import {ReplayPlayPauseBar} from 'sentry/components/replays/replayController';
```

Comment:
Looks like 3 things were exported, but this is the only import. Also looks like at least `StyledScrubber` was copied over.

I don't love exporting all the little bits and then recombining them into the same unit... it looks like this could be it's own thing as one export and then imported into replayController and here.

"TimeAndScrubberGrid" is a reasonable name for that new file, i'd use that for the main export, and rename the existing styled thing to be just Grid or something simpler.
```
export default TimeAndScrubberGrid = function(props: Props) {
  return (
    <Container>
      <Grid>
        ...
      </Grid>
    </Container>
  );
};
const Grid = styled('div')`
  ...
  grid-template-areas:
    '. timeline .'
    'currentTime scrubber duration';
  ...
`;
```

Reply:
i think it's better to start with sharing, and then when things diverge we can choose how to fork things to match the case. like it's possible to add props, decompose into smaller parts and share the smaller bit, or fall through to copy paste depending on the situation